src/python/pants/jvm/resolve/lockfile_metadata.py

```diff
@@ -76,7 +77,21 @@ class JVMLockfileMetadataV1(JVMLockfileMetadata):
     def from_artifact_requirements(
         cls, requirements: Iterable[ArtifactRequirement]
     ) -> JVMLockfileMetadataV1:
-        return cls(FrozenOrderedSet(sorted(i.to_metadata_str() for i in requirements)))
+        reqs = list(requirements)
+        req_sources = [
+            dataclasses.replace(
+                i, coordinate=dataclasses.replace(i.coordinate, classifier="sources")
+            )
+            for i in reqs
+        ]
+
+        extended = []
+        for i in reqs:
+            extended.append(i)
+        for i in req_sources:
+            extended.append(i)
+
+        return cls(FrozenOrderedSet(sorted(i.to_metadata_str() for i in extended)))
```

Comment:
Again if enabling source jars it would be added to the lockfile and lockfile metadata as well